### DISCUSSION TITLE
Component OR matching in view

### DISCUSSION THREAD

**Discussion by jdtec01** [skypjack/entt#765] [Q&A] [answered]
Hi,

Is it possible to create a view with a query for 'any of' / OR a list of components? eg. atm I am creating two identical views to do the same work, except one takes a `PathAgent_FollowingPath` and the other takes a `PathAgent_PendingPath`:

```
{auto view = Registry ().view<MovementGoal, const PathAgentGoal, const PathAgent_FollowingPath> ();
    for (auto [unitEntity, movementGoal, pathGoal]: view.each ())
    {
      movementGoal.Value = pathGoal.Value;
    }
  }
  {auto view = Registry ().view<MovementGoal, const PathAgentGoal, const PathAgent_PendingPath> ();
    for (auto [unitEntity, movementGoal, pathGoal]: view.each ())
    {
      movementGoal.Value = pathGoal.Value;
    }
  }
```
  
  But what I'd really like is something like this that says you can have 'any' or either of the following components for the query to match. Basically a component OR :
  
 ```
 {auto view = Registry ().view<MovementGoal, const PathAgentGoal> (entt::any<PathAgent_FollowingPath, PathAgent_PendingPath>);
    for (auto [unitEntity, movementGoal, pathGoal]: view.each ())
    {
      movementGoal.Value = pathGoal.Value;
    }
  }
```

Is it possible to do something like this in entt?

**Comment by skypjack** [comment, marked as answer]
Uhm, no, actually it's not possible, mainly because there is no way to optimize it more than what you would do on the user side.
There is a plan to add policies to views for this kind of user side filters but they are not there yet. In any case, they'll be nothing more than syntactic sugar.

**Reply by jdtec01** [reply]
OK, thanks for the answer!
